subcommand/webhook-cert-manager/command_test.go

```diff
@@ -0,0 +1,275 @@
+)
+
+func TestRun_FlagValidation(t *testing.T) {
+	t.Parallel()
+
+	cases := []struct {
+		flags  []string
+		expErr string
+	}{
+		{
+			flags:  nil,
+			expErr: "-config-file must be set",
+		},
+	}
+
+	for _, c := range cases {
+		t.Run(c.expErr, func(tt *testing.T) {
+			ui := cli.NewMockUi()
+			cmd := Command{
+				UI: ui,
+			}
+			exitCode := cmd.Run(c.flags)
+			require.Equal(tt, 1, exitCode, ui.ErrorWriter.String())
+			require.Contains(tt, ui.ErrorWriter.String(), c.expErr)
+		})
+	}
+}
+
+func TestRun_SecretDoesNotExist(t *testing.T) {
+	t.Parallel()
+	secretOne := "secret-deploy-1"
+	secretTwo := "secret-deploy-2"
+
+	webhookConfigOne := "webhookOne"
+	webhookConfigTwo := "webhookTwo"
+
+	caBundleOne := []byte("bootstrapped-CA-one")
+	caBundleTwo := []byte("bootstrapped-CA-two")
+
+	webhookOne := &admissionv1beta1.MutatingWebhookConfiguration{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: webhookConfigOne,
+		},
+		Webhooks: []admissionv1beta1.MutatingWebhook{
+			{
+				Name: "webhookOne-under-test",
+				ClientConfig: admissionv1beta1.WebhookClientConfig{
+					CABundle: caBundleOne,
+				},
+			},
+		},
+	}
+	webhookTwo := &admissionv1beta1.MutatingWebhookConfiguration{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: webhookConfigTwo,
+		},
+		Webhooks: []admissionv1beta1.MutatingWebhook{
+			{
+				Name: "webhookOne-under-test",
+				ClientConfig: admissionv1beta1.WebhookClientConfig{
+					CABundle: caBundleTwo,
+				},
+			},
+		},
+	}
+
+	k8s := fake.NewSimpleClientset(webhookOne, webhookTwo)
+	ui := cli.NewMockUi()
+	cmd := Command{
+		UI:        ui,
+		clientset: k8s,
+	}
+	cmd.init()
+
+	file, err := ioutil.TempFile("", "config.json")
+	require.NoError(t, err)
+	defer os.Remove(file.Name())
+
+	_, err = file.Write([]byte(configFile))
+	require.NoError(t, err)
+
+	exitCh := runCommandAsynchronously(&cmd, []string{
+		"-config-file", file.Name(),
+	})
+	defer stopCommand(t, &cmd, exitCh)
+
+	timer := &retry.Timer{Timeout: 10 * time.Second, Wait: 500 * time.Millisecond}
+	retry.RunWith(timer, t, func(r *retry.R) {
+		secret1, err := k8s.CoreV1().Secrets("default").Get(context.TODO(), secretOne, metav1.GetOptions{})
+		require.NoError(r, err)
+		require.Equal(r, secret1.Type, v1.SecretTypeTLS)
+
+		secret2, err := k8s.CoreV1().Secrets("default").Get(context.TODO(), secretTwo, metav1.GetOptions{})
+		require.NoError(r, err)
+		require.Equal(r, secret2.Type, v1.SecretTypeTLS)
+
+		webhookConfig1, err := k8s.AdmissionregistrationV1beta1().MutatingWebhookConfigurations().Get(context.TODO(), webhookConfigOne, metav1.GetOptions{})
+		require.NoError(t, err)
+		require.NotEqual(r, webhookConfig1.Webhooks[0].ClientConfig.CABundle, caBundleOne)
+
+		webhookConfig2, err := k8s.AdmissionregistrationV1beta1().MutatingWebhookConfigurations().Get(context.TODO(), webhookConfigTwo, metav1.GetOptions{})
+		require.NoError(t, err)
+		require.NotEqual(r, webhookConfig2.Webhooks[0].ClientConfig.CABundle, caBundleTwo)
```

Comment:
```suggestion
		require.NotEqual(r, webhookConfig2.Webhooks[1].ClientConfig.CABundle, caBundleTwo)
```

Reply:
is there any way we can inject a fake cert generator? this way we can write stronger assertions on the value of the caBundle. This would pass even if caBundle was updated to an empty string.